pybossa/view/projects.py

```diff
@@ -3236,14 +3236,26 @@ def ext_config(short_name):
                     flash(gettext('Please correct the errors', 'error'))
                 ext_conf[form_name] = {k: v for k, v in six.iteritems(form.data) if v}
                 ext_conf[form_name].pop('csrf_token', None)     #Fflask-wtf v0.14.2 issue 102
-                project.info['ext_config'] = ext_conf
-                project_repo.save(project)
-                sanitize_project, _ = sanitize_project_owner(project, owner, current_user, ps)
 
-                current_app.logger.info('Project id {} external configurations set. {} {}'.format(
-                    project.id, form_name, form.data))
-                flash(gettext('Configuration for {} was updated').format(display), 'success')
+                target_bucket = ext_conf.get('gigwork_poller', {}).get('target_bucket')
+                if not target_bucket:
+                    ext_conf.pop('gigwork_poller', None)
+
+                hdfs_path = ext_conf.get('hdfs', {}).get('path')
+                if not hdfs_path:
+                    ext_conf.pop('hdfs', None)
 
+                if not ext_conf and current_app.config.get('PRIVATE_INSTANCE'):
+                    flash(gettext('At least one target location must be provided. {} was not updated').format(display), 'error')
```

Comment:
UI specifies target location as response file & consensus location. As the error won' t highlight to fields(w/ red border), in order to easily relate the error, the error message could be following or similar
```
At least one response file & consensus location must be provided.
```